api/tests/Security/Voter/CampRoleVoterTest.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+namespace App\Tests\Security\Voter;
+
+use App\Entity\BaseEntity;
+use App\Entity\Camp;
+use App\Entity\CampCollaboration;
+use App\Entity\Period;
+use App\Entity\User;
+use App\Security\Voter\CampRoleVoter;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+
+/**
+ * @internal
+ */
+class CampRoleVoterTest extends TestCase {
+    private CampRoleVoter $voter;
+    private TokenInterface|MockObject $token;
+
+    public function setUp(): void {
+        parent::setUp();
+        $this->token = $this->createMock(TokenInterface::class);
+        $this->voter = new CampRoleVoter();
+    }
+
+    public function testDoesntVoteWhenAttributeWrong() {
+        // given
+
+        // when
+        $result = $this->voter->vote($this->token, new Period(), ['CAMP_SUPPORTER']);
+
+        // then
+        $this->assertEquals(CampRoleVoter::ACCESS_ABSTAIN, $result);
```

Comment:
I would use the VoterInterface class, because that's where the constant ist defined, instead of CampRoleVoter.
It also indicates that the value is defined by the framework, and not by us.